src/Bramus/Router/Router.php

```diff
@@ -373,7 +440,63 @@ private function handle($routes, $quitAfterRun = false)
         return $numHandled;
     }
 
-    private function invoke($fn, $params = [])
+    /**
+     * Set sub routes uri prefix.
+     * @param $prefix string A router prefix such as /admin
+     * @return $this
+     */
+    public function prefix($prefix)
+    {
+        $router = clone $this;
+        $router->setSubPrefix($prefix);
+
+        return $router;
```

Comment:
I see here (and also elsewhere in the code) that you always clone the router. Is there a specific reason for this? Couldn't it be done by directly returning `$this`?

For example:

```php
public function prefix($prefix)
{
    $this->setSubPrefix($prefix);
    return $this;    
}
```

Reply:
Clone can generate new contexts. This is mainly for coding convenience.